File: GeneratedEntityReaderClasses/Components/GameAreaEffectComponent.cs
namespace GeneratedNoitaClasses;
public class GameAreaEffectComponent : NoitaComponentBase,  iNoitaType<GameAreaEffectComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat radius { get; set; } = new();
	public NoitaString collide_with_tag { get; set; } = new();
	public NoitaInt frame_length { get; set; } = new();
	public NoitaVector<NoitaString> game_effect_entitities { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		radius.Read(s);
		collide_with_tag.Read(s);
		frame_length.Read(s);
		game_effect_entitities.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		radius.Write(s);
		collide_with_tag.Write(s);
		frame_length.Write(s);
		game_effect_entitities.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/GameAreaEffectComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaVector
from BasicNoitaTypes import NoitaString

from NoitaStream import NoitaStream
@dataclass
class GameAreaEffectComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    radius:NoitaFloat = field(default_factory=NoitaFloat)
    collide_with_tag:NoitaString = field(default_factory=NoitaString)
    frame_length:NoitaInt = field(default_factory=NoitaInt)
    game_effect_entitities:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaString))
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.radius.Read(s)
        self.collide_with_tag.Read(s)
        self.frame_length.Read(s)
        self.game_effect_entitities.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.radius.Write(s)
        self.collide_with_tag.Write(s)
        self.frame_length.Write(s)
        self.game_effect_entitities.Write(s)

